packages/amplication-server/src/core/assistant/assistant.service.spec.ts

```diff
@@ -0,0 +1,521 @@
+import { Entity } from "../../models";
+
+const EXAMPLE_CHAT_OPENAI_KEY = "EXAMPLE_CHAT_OPENAI_KEY";
+const EXAMPLE_WORKSPACE_ID = "EXAMPLE_WORKSPACE_ID";
+const EXAMPLE_PROJECT_ID = "EXAMPLE_PROJECT_ID";
+const EXAMPLE_RESOURCE_ID = "EXAMPLE_RESOURCE_ID";
+const EXAMPLE_THREAD_ID = "EXAMPLE_THREAD_ID";
+const EXAMPLE_USER_ID = "EXAMPLE_USER_ID";
+
+const EXAMPLE_ENTITY: Entity = {
+  id: "exampleEntityId",
+  createdAt: new Date(),
+  updatedAt: new Date(),
+  resourceId: "exampleResourceId",
+  name: "exampleName",
+  displayName: "exampleDisplayName",
+  pluralDisplayName: "examplePluralDisplayName",
+  customAttributes: "exampleCustomAttributes",
+  lockedByUserId: EXAMPLE_USER_ID,
+};
+
+const EXAMPLE_ASSISTANT_CONTEXT: AssistantContext = {
+  user: {
+    workspace: {
+      allowLLMFeatures: true,
+      id: EXAMPLE_WORKSPACE_ID,
+      createdAt: new Date(),
+      updatedAt: new Date(),
+      name: "",
+    },
+    id: EXAMPLE_USER_ID,
+    createdAt: new Date(),
+    updatedAt: new Date(),
+    isOwner: true,
+  },
+  workspaceId: EXAMPLE_WORKSPACE_ID,
+};
+
+const pubSubPublishMock = jest.fn();
+
+const mockGraphqlSubscriptionKafkaService = {
+  getPubSub: jest.fn(() => ({
+    publish: pubSubPublishMock,
+    asyncIterator: jest.fn(),
+  })),
+};
+
+const entityServiceCreateOneEntityMock = jest.fn(() => EXAMPLE_ENTITY);
+const entityServiceCreateFieldByDisplayNameMock = jest.fn();
+const projectServiceCreateProjectMock = jest.fn();
+const resourceServiceCreateServiceWithDefaultSettingsMock = jest.fn();
+const moduleServiceCreateMock = jest.fn();
+const pluginInstallationServiceCreateMock = jest.fn(() => ({
+  id: "examplePluginId",
+}));
+const pluginCatalogServiceGetPluginWithLatestVersionMock = jest.fn(() => ({
+  id: "examplePluginId",
+  pluginId: "examplePluginId",
+  name: "exampleName",
+  description: "exampleDescription",
+  repo: "exampleRepo",
+  npm: "exampleNpm",
+  icon: "exampleIcon",
+  github: "exampleGithub",
+  website: "exampleWebsite",
+  categories: ["exampleCategory1", "exampleCategory2"],
+  type: "exampleType",
+  taggedVersions: {},
+  versions: [
+    {
+      id: "exampleVersionId",
+      pluginId: "examplePluginId",
+      deprecated: false,
+      isLatest: true,
+      version: "exampleVersion",
+      settings: {},
+      configurations: {},
+    },
+  ],
+}));
+
+const resourceServiceResourcesMock = jest.fn();
+const moduleServiceFindManyMock = jest.fn();
+const moduleDtoServiceCreateMock = jest.fn();
+const moduleDtoServiceCreateEnumMock = jest.fn();
+const moduleDtoServiceFindManyMock = jest.fn();
+
+const moduleActionServiceCreateMock = jest.fn();
+const projectServiceCommitMock = jest.fn();
+const projectServiceGetPendingChangesMock = jest.fn();
+const pluginCatalogServiceGetPluginsMock = jest.fn();
+const moduleActionServiceFindManyMock = jest.fn();
+const entityServiceEntitiesMock = jest.fn();
+describe("AssistantService", () => {
+  let service: AssistantService;
+
+  beforeEach(async () => {
```

Comment:
For nestjs modules initialisation, using `beforeAll` will greatly improve the tests performances as it will use way less memory.

Only mocks, fakes and stub expectations should be in the beforeEach

Reply:
thanks, good to know